sherpa/astro/data.py

```diff
@@ -1761,6 +1771,44 @@ def get_specresp(self, filter=False):
 
         return newarf
 
+    def get_full_response(self, pileup_model=None):
+        """Calculate the response for the dataset.
+
+        Unlike `get_response`, which returns a single response, this function
+        returns all responses for datasets that have multiple responses set
+        and it offers the possibility to include a pile-up model.
+
+        Parameters
+        ----------
+        pileup_model : None or a `sherpa.astro.models.JDPileup` instance
+            If a pileup model shall be included in the return, then it needs
+            to be passed in.
+
+        Returns
+        -------
+        response
+           The return value depends on whether an ARF, RMF, or pile up
+           model has been associated with the data set.
+
+        See Also
+        --------
+        get_response, get_arf, get_rmf
+        """
+        # import is here because sherpa.astro.instrument depends on
+        # sherpa.astro.data. Importing here instead of on the top
+        # avoids a circular import.
+        from sherpa.astro import instrument
```

Comment:
I'm not a big fan of how this introduces a circular dependency (only worked around by moving the import into the method), but conceptually I think this functionality definitely belongs into the data class. Another facet of the discussion in #1061. This would go away if we did not distinguish between data and instrument ARF and RMf - a distinction that's not clear to me anyway.